asyncpg/connect_utils.py

```diff
@@ -789,15 +797,30 @@ def data_received(self, data):
                     'rejected SSL upgrade'.format(
                         host=self.host, port=self.port)))
 
-    def connection_lost(self, exc):
+    def connection_lost(self, exc: typing.Optional[Exception]) -> None:
         if not self.on_data.done():
             if exc is None:
                 exc = ConnectionError('unexpected connection_lost() call')
             self.on_data.set_exception(exc)
 
 
-async def _create_ssl_connection(protocol_factory, host, port, *,
-                                 loop, ssl_context, ssl_is_advisory=False):
+_ProctolFactoryR = typing.TypeVar(
+    "_ProctolFactoryR", bound=asyncio.protocols.Protocol
+)
+
+
+async def _create_ssl_connection(
+    # TODO: The return type is a specific combination of subclasses of
+    # asyncio.protocols.Protocol that we can't express. For now, having the
+    # return type be dependent on signature of the factory is an improvement
+    protocol_factory: "Callable[[], _ProctolFactoryR]",
```

Comment:
```suggestion
    protocol_factory: Callable[[], _ProctolFactoryR],
```

Reply:
For compatibility you want `typing.Callable`, not `collections.abc.Callable`

Reply:
In later versions `collections` is preferred though. Isn't the use of stringified annotations the way to get compatibility?

Reply:
If you add `from __future__ import annotations` to the top of the file, you won't have to use strings in the annotation (and importing from `collections.abc` will also work)

Reply:
Also fine with me, @elprans what has your preference?

Reply:
Yeah, do that please.

Reply:
Fixed!